src/components/organisms/LocalSearch.tsx

```diff
@@ -0,0 +1,932 @@
+            navigateToNextItem('previous')
+          }
+          break
+        case 'Escape':
+          onSearchClose()
+          break
+        case 'F':
+        case 'f':
+          if (event.ctrlKey && onReplaceToggle) {
+            onReplaceToggle(false)
+          }
+          break
+      }
+    },
+    [
+      onReplaceToggle,
+      showingReplace,
+      navigateToNextItem,
+      getNumberOfTextAreaRows,
+      onSearchClose,
+      addNewlineToSearchValue,
+      toggleRegexSearch,
+      toggleCaseSensitiveSearch,
+    ]
+  )
+
+  useEffect(() => {
+    function onContentChange(
+      _: CodeMirror.Editor,
+      changes: EditorChangeLinkedList
+    ) {
+      if (changes.origin == '@ignore') {
+        return
+      }
+
+      // restore replace focus before updating results otherwise it might render
+      // out of focus because value don't have time to propagate
+      if (focusingReplace) {
+        setFocusingReplace(false)
+      }
+      updateSearchResults(
+        searchValue,
+        regexSearch,
+        caseSensitiveSearch,
+        true,
+        false
+      )
+    }
+    codeMirror.on('change', onContentChange)
+    return () => {
+      codeMirror.off('change', onContentChange)
+    }
+  }, [
+    caseSensitiveSearch,
+    codeMirror,
+    focusingReplace,
+    getSearchRegex,
+    regexSearch,
+    searchValue,
+    updateSearchResults,
+  ])
+
+  useEffectOnce(() => {
+    updateSearchResults(searchValue, regexSearch, caseSensitiveSearch)
+    if (showingReplace && searchValue.length !== 0) {
+      setFocusingReplace(true)
+    } else {
+      focusSearchTextAreaInput(searchValue.length, true)
+    }
+  })
+
+  useEffect(() => {
+    if (loadingResults) {
+      updateSearchResults(
+        searchValue,
+        regexSearch,
+        caseSensitiveSearch,
+        false,
+        false
+      )
+      if (onUpdateSearchOptions != null) {
+        onUpdateSearchOptions({
+          caseSensitiveSearch: caseSensitiveSearch,
+          regexSearch: regexSearch,
+        })
+      }
+      setLoadingResults(false)
+    }
+  }, [
+    caseSensitiveSearch,
+    focusingReplace,
+    loadingResults,
+    onUpdateSearchOptions,
+    regexSearch,
+    searchValue,
+    updateSearchResults,
+  ])
+
+  useDebounce(
+    () => {
+      setLoadingResults(true)
+    },
+    SEARCH_DEBOUNCE_TIMEOUT,
+    [searchValue, regexSearch, caseSensitiveSearch]
+  )
+
+  return (
+    <LocalSearchContainer>
+      <SearchResultItem>
+        <LocalSearchInputLeft>
+          <LocalSearchIcon
+            className={getNumberOfTextAreaRows != 1 ? 'alignStart' : ''}
+          >
+            <Icon path={mdiMagnify} />
+          </LocalSearchIcon>
+          <textarea
+            onClick={focusSearchTextAreaInput}
+            rows={getNumberOfTextAreaRows}
+            value={searchValue}
+            onInput={updateSearchValue}
+            onChange={updateSearchValue}
+            // onChange={() => undefined}
```

Comment:
Forgot to comment on this.

So the onChange and onInput events are kind of weird in react. Googling (possibly outdated) says in React this works the same (no difference between events), in general, there is a difference, since one emits on any input change, and the other only on acknowledged change.

As for how it behaves in React by experiment:
1. Only on change, it works ok but when a user adds newline and then deletes it - the event is not fired on the deletion of the newline character (not really what we want here)
2. Only on input, works fine (fires like it should, even on newline addition/removal but React then gives a warning that we have bound value to textarea and no onChange event is bound
3. The warning can be removed by binding it to undefined (commented out here) or just binding it to the same function as `onInput` - works OK, but this might not be the most optimized version since it's called twice (in some cases)

Reply:
the second option looks better.

Reply:
Yeah, it's the most correct version, but to get rid of React warnings we should either keep it as it is now or add a dummy undefined `onChange` (the commented out one), please update it as you see fit before merging or let me know what is the resolution.